Runtime/World/World.cpp

```diff
@@ -54,6 +48,19 @@ namespace Spartan
 		SUBSCRIBE_TO_EVENT(Event_World_Resolve, [this](Variant) { m_isDirty = true; });
 		SUBSCRIBE_TO_EVENT(Event_World_Stop,	[this](Variant)	{ m_state = Idle; });
 		SUBSCRIBE_TO_EVENT(Event_World_Start,	[this](Variant)	{ m_state = Ticking; });
+
+        // Hardcoded types make it harder to write new components
```

Comment:
Hardcoded types mean that when a coder writes a new component, he'll have to visit all of this places and updated them. Do you think we can avoid them somehow ?